ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsTask.java

```diff
@@ -118,11 +118,15 @@ public String getName() {
     private boolean isMissingAcidState = false;
     private BasicStatsWork work;
     private boolean followedColStats1;
+    private boolean isBasicStatProvided;
+    private Map<String, String> providedBasicStats;
 
-    public BasicStatsProcessor(Partish partish, BasicStatsWork work, HiveConf conf, boolean followedColStats2) {
+    public BasicStatsProcessor(Table table, Partish partish, BasicStatsWork work, boolean followedColStats2) {
```

Comment:
you seem to have added a totally independent conditional  logic to this class; wouldn't it be easier to simply introduce a new `Processor` class for the purpose your are targeting?

Reply:
Per our discussion, I moved my changes to the `BasicStatsNoJobTask`.